docs/topics/Automated-Checks.md

```diff
@@ -0,0 +1,259 @@
+# Automated Checks
+
+The [](DataWorkerService.md) has numerous responsibilities, one of them being a data processing step known as automated
+checks. These checks are responsible for processing various portions of data depending on the current processing step
+for a particular piece of data.
+
+## Core Principals
+
+When designing this system, we did so with the following principles in mind:
+
+1. Human reviewers have authority over whether an entity is `Verified` or `Rejected`. As such, the system will never automatically assign these designations.
+2. The automatic application of the `PreRejected` status must be as accurate as possible, based on concrete rules.
+3. The process must be as transparent as possible. As such, the system tracks all changes to entities in the `audit` tables. Additionally, all entities have a `RejectionReason` enum which defines a combination of reasons why it was marked as rejected by either the system or human reviewer.
+4. Do not include entities which are not `Verified` in the tournament rating algorithm.
+    * This provides an added benefit of ensuring all generated statistics are valid. Even with manually submitted data, humans make mistakes. If unverified data is introduced into the rating & statistics systems, users will notice invalid statistics and the rating ladder itself will not be completely accurate.
+
+## Entities
+
+The following entities are part of this processing pipeline:
+
+* `Tournaments`
+* `Matches`
+* `Games`
+* `GameScores`
+
+## Statuses
+
+Each entity has `VerificationStatus`, `ProcessingStatus`, and `RejectionReason` fields. These fields are referenced and changed by the DataWorkerService as they move through the processing flow.
+
+### `VerificationStatus`
+
+Each entity shares the same `VerificationStatus` type. This type contains the following statuses:
+
+* `None`: The entity has yet to be processed automatically.
+* `PreRejected`: Based on the system's rules, this entity should be rejected.
+* `PreVerified`: The system did not find anything wrong, awaiting human review.
+* `Rejected`: A human marked this entity as rejected.
+* `Verified`: A human marked this entity as verified.
+
+### `ProcessingStatus`
+
+Each entity has a unique `ProcessingStatus` type associated with it. This flag is self-explanatory: it indicates how far along an entity is in the processing pipeline.
+
+For example, consider `TournamentProcessingStatus`:
+
+1. `NeedsApproval`: The tournament is submitted but waiting approval from a verifier.
+2. `NeedsMatchData`: Match data needs to be fetched via the osu! API.
+3. `NeedsAutomationChecks`: The tournament, and all of its children, are awaiting automation checks.
+4. `NeedsVerification`: Awaiting human review
+5. `NeedsStatCalculation`: After human review, process statistics (must be complete before it is eligible for inclusion in the rating system).
```

Comment:
something like this?
```suggestion
5. `NeedsStatCalculation`: Statistics, such as match scores, must be processed and stored (so that the matches are ready to be inputted into the rating algorithm).
```

Reply:
Will go with
> Some statistics, such as game placements, must be processed and stored as the processor requires them to function.